docs/site/components/markdown/markdown-service.js

```diff
@@ -0,0 +1,28 @@
+angular
+  .module('gcloud.markdown')
+  .factory('mdConverter', function() {
+    return new showdown.Converter({
+      extensions: [function () {
+        return [{
+          type: 'lang',
+          // unescaped version \[([^\]]+)\]\(([\/|\#][^)]+)\)
+          regex: '\\[([^\\]]+)\\]\\(([\\/|\\#][^)]+)\\)',
+          replace: function (match, text, href) {
+            var isDeeplink = /^\#/.test(href);
+            var attribute;
+
+            if (isDeeplink) {
+              href = href.split(/[\#|\-]/).join('');
+              attribute = 'href="" ng-click="setSection(\'' + href + '\')"';
```

Comment:
Yeah, I think that's doable, before this was a provider so I didn't have access to `$location` but since changing it, we could probably just set the actual link to the href now.